app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt

```diff
@@ -0,0 +1,122 @@
+
+@Composable
+fun QuickFixSlidingWindow(
+    isVisible: Boolean,
+    onDismiss: () -> Unit,
+    content: @Composable () -> Unit
+) {
+  BoxWithConstraints(
+      modifier = Modifier.testTag("QuickFixSlidingWindow"),
+  ) {
+    val density = LocalDensity.current
+    val screenWidthPx = constraints.maxWidth.toFloat()
+    val screenWidthDp = with(density) { constraints.maxWidth.toDp() }
+
+    // Calculate window width (90% of the screen width) and the left gap (10%)
+    val windowWidthDp = screenWidthDp * 0.9f
+    val windowWidthPx = with(density) { windowWidthDp.toPx() }
+    val gapWidthPx = screenWidthPx - windowWidthPx
+
+    // Dismiss threshold (e.g., 25% of the window width)
+    val dismissThreshold = windowWidthPx * 0.25f
+
+    // State for drag offset
+    var dragOffsetX by remember { mutableStateOf(0f) }
+
+    // Animatable for the window's horizontal position
+    val animatableOffsetX = remember { Animatable(screenWidthPx) }
+
+    // Coroutine scope for launching animations
+    val coroutineScope = rememberCoroutineScope()
+
+    // Launch animation when isVisible changes
+    LaunchedEffect(isVisible) {
+      if (isVisible) {
+        // Reset drag offset
+        dragOffsetX = 0f
+        // Animate in from off-screen to the gap position
+        animatableOffsetX.snapTo(screenWidthPx)
+        animatableOffsetX.animateTo(
+            targetValue = gapWidthPx, animationSpec = tween(durationMillis = 100))
+      } else {
+        // Animate out to off-screen
+        animatableOffsetX.animateTo(
+            targetValue = screenWidthPx, animationSpec = tween(durationMillis = 100))
+      }
+    }
+
+    Box(Modifier.fillMaxSize()) {
+      if (isVisible || animatableOffsetX.isRunning) {
+        // Show the window only when it's visible or animating
+        Box(
+            modifier =
+                Modifier.fillMaxHeight()
+                    .testTag("QuickFixSlidingWindowContent")
+                    .width(windowWidthDp)
+                    .offset {
+                      IntOffset(x = (animatableOffsetX.value + dragOffsetX).roundToInt(), y = 0)
+                    }
+                    .pointerInput(Unit) {
+                      detectHorizontalDragGestures(
+                          onDragEnd = {
+                            if (dragOffsetX > dismissThreshold) {
+                              onDismiss()
+                            } else {
+                              coroutineScope.launch {
+                                animatableOffsetX.animateTo(
+                                    targetValue = gapWidthPx,
+                                    animationSpec = tween(durationMillis = 100))
+                              }
+                              dragOffsetX = 0f
+                            }
+                          },
+                          onHorizontalDrag = { _, dragAmount ->
+                            dragOffsetX = (dragOffsetX + dragAmount).coerceAtLeast(0f)
```

Comment:
you can add a minDragDistance to ignore tiny, unintentional drags

Reply:
it already handles this 😄